Source/WebCore/rendering/svg/RenderSVGResourcePaintServer.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2023 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+#include "Color.h"
+#include "RenderSVGResourceContainer.h"
+#include <variant>
+
+namespace WebCore {
+
+class GraphicsContext;
+class RenderSVGShape;
+
+class RenderSVGResourcePaintServer : public RenderSVGResourceContainer {
+    WTF_MAKE_ISO_ALLOCATED(RenderSVGResourcePaintServer);
+public:
+    virtual ~RenderSVGResourcePaintServer();
```

Comment:
It seems we sometimes use "= default" for dtors in headers but here we use it in the cpp. I guess we should have a consistent style, I prefer to have it in the header.

Reply:
I always keep virtual destructors in the .cpp file - no strong opinion here. Need to refresh my mind about that topic :-)